### PR TITLE
Constexprification, NSDMIifying, and other code cleanup

### PR DESCRIPTION
**Pull request by DrItanium**
I was going to do this in multiple different pull requests but this is the result of looking at "expensive" functions and cleaning up corresponding classes.

### CODE REVIEW

File: src/utils/graphicsutils.cpp [hunk 1]
```diff
@@ -73,51 +75,30 @@ void GraphicsUtils::distanceFromLine(double cx, double cy, double ax, double ay,
...
-	pd->d = (current.y() - initial.y()) * (current.y() - initial.y());
-	pds.append(pd);
-
-	pd = new PD;
-	pd->p.setX(initial.x());
-	pd->p.setY(current.y());
-	pd->d = (current.x() - initial.x()) * (current.x() - initial.x());
-	pds.append(pd);
-
+    std::vector<PD> pds;
```
Comment (DrItanium) [inline src/utils/graphicsutils.cpp]: I would have used QList but it does not provide the ability to construct objects in place.

File: src/fsvgrenderer.cpp [hunk 2]
```diff
@@ -342,19 +335,22 @@ void FSvgRenderer::initLegInfoAux(QDomElement & element, const LoadInfo & loadIn
 bool FSvgRenderer::initLegInfoAux(QDomElement & element, ConnectorInfo * connectorInfo)
 {
 	bool ok;
-	double sw = element.attribute("stroke-width").toDouble(&ok);
+    auto extractDoubleFromAttribute = [&element, &ok](const QString& attribute) {
+        return element.attribute(attribute).toDouble(&ok); 
+    };
```
Comment (KjellMorgenstern) [inline src/fsvgrenderer.cpp]:
```suggestion
    };
```

File: src/fsvgrenderer.h [hunk 3]
```diff
@@ -53,12 +53,11 @@ struct LoadInfo {
...
-	bool findNonConnectors;
-	bool parsePaths;
+	bool findNonConnectors = false;
+	bool parsePaths = false;;
 
-	LoadInfo() {
-		findNonConnectors = parsePaths = false;
-	}
+	LoadInfo() = default;
+    LoadInfo(const QString& file, bool _findNonConnector = false) : filename(file), findNonConnectors(_findNonConnector) { }
```
Comment (KjellMorgenstern) [inline src/fsvgrenderer.h]: Tabs

File: src/partsbinpalette/svgiconwidget.cpp [hunk 4]
```diff
@@ -216,11 +216,8 @@ void SvgIconWidget::setupImage(bool plural, ViewLayer::ViewID viewID)
 		delete icon;
 	}
 
-	m_pixmapItem = new SvgIconPixmapItem(pixmap, this);
-	m_pixmapItem->setPlural(plural);
-
-	m_pixmapItem->setFlags(0);
-	m_pixmapItem->setPos(0, 0);
+    /// @todo make sure this doesn't leak if this function is called multiple times on the same object.
```
Comment (KjellMorgenstern) [inline src/partsbinpalette/svgiconwidget.cpp]: Tabs

File: src/utils/graphicsutils.cpp [hunk 5]
```diff
@@ -25,14 +25,8 @@ along with Fritzing.  If not, see <http://www.gnu.org/licenses/>.
...
-const double GraphicsUtils::SVGDPI = 90;
-const double GraphicsUtils::InchesPerMeter = 39.370078;
-const double GraphicsUtils::StandardSchematicSeparationMils = 295.275591;   // 7.5mm
-const double GraphicsUtils::StandardSchematicSeparation10thinMils = 100;   // 0.1 inches
-
+#include <vector>
+#include <algorithm>
 
 void GraphicsUtils::distanceFromLine(double cx, double cy, double ax, double ay, double bx, double by,
                                      double & dx, double & dy, double &distanceSegment, bool & atEndpoint)
```
Comment (KjellMorgenstern) [inline src/utils/graphicsutils.cpp]: The implementation of this method was moved to the header file?

**Pull request merged**